enhancements/network/dpu-ovnkube-operator.md

```diff
@@ -0,0 +1,186 @@
+the normal user applications run on, is composed of the X86 servers.
+
+The OVN-Kubernetes components are spread over the two clusters. On the tenant
+cluster side, the Cluster Network Operator is in charge of the management of the
+ovn-kube components. On the infra cluster side, We propose to create a new
+operator to be responsible for the life-cycle management of ovn-kube components.
+
+## Motivation
+
+Normally, the Cluster Network Operator watches the Network customer resource,
+then provision the ovn-kube components to the Openshift cluster.
+
+In the two-cluster design, some of the ovn-kube components have to run on the
+SmartNIC, but they are still part of the ovn-kube cluster running in the tenant
+cluster. To allow the ovn-kube components on different clusters to talk with
+each other, we need to ensure all the necessary configs and secrets are aligned
+with each other.
+
+### Goals
+
+- Design a new operator that
+  - manages the ovn-kube components that serve the tenant cluster.
+  - enable the swichedev mode on BlueField card.
+- Define the ovn-kube upgrade mechanism in such a deployment.
+
+### Non-Goals
+
+Support for network plugins other than OVN Kubernetes.
+This enhancement does not address the two-cluster installation.
+
+## Pre-requisites
+
+- This operator needs to access the apiserver of the tenant cluster, we need to
+make sure that the pods network in infra cluster can connect to the apiserver of
+the tenant cluster in some way.
+- One custom MachineConfigPool will be created for the BlueField nodes on which
+hardware offloading needs to be enabled.
+
+## Proposal
+
+We will create a namespaced CRD OVNKubeConfig.smartnic.openshift.io for the
+DPU OVNKube Operator. One custom resource of this CRD shall be created for
+the tenant cluster. The operator will take this custom resource as input to
+render and create the objects of ovnkube smart-nic components. The operator will
+also create a MachineConfig CR which injects a systemd service to enable
+switchdev mode on BF-2 node.
+
+### User Stories
+
+As a cluster admin I want to manage the ovn-kubernetes components in the infra
+cluster, and these components can be controlled by the ovn-kubernetes control plane in the tenant cluster.
+
+### Risks and Mitigations
+
+## Design Details
+
+The DPU OVNKube Operator will fetch the necessary information from the
+tenant cluster, then create the following objects in the infra cluster:
+- The ovnkube-node DaemonSet running in the SmartNIC mode
+- The ServiceAccount and corresponding RBAC definition
+- The ConfigMap that contains ​​ovnkube-config of the tenant cluster
+- The ComfgMap and Secret that contains the CA and signed certificate of the
+- tenant cluster OVN
+- The ConfigMap CA and signed certificate of the tenant cluster API server
+
+The DPU OVNKube Operator will be responsible for enable the switchdev mode on
+BF2 cards. We assume all the BF2 cards will be put into a custom
+MachineConfigPool,
```

Comment:
comma should be a period, or else there's something missing at the end?

Who is responsible for the custom MachineConfigPool? When are they supposed to create it?

Reply:
> Who is responsible for the custom MachineConfigPool? When are they supposed to create it?

That is a good question. In my assumption, the cluster-admin shall label the nodes and create the MCP. WDYT?